pkg/kv/kvserver/store.go

```diff
@@ -2956,10 +2971,19 @@ func (s *Store) Capacity(ctx context.Context, useCached bool) (roachpb.StoreCapa
 	var logicalBytes int64
 	var totalQueriesPerSecond float64
 	var totalWritesPerSecond float64
+	var l0SublevelsMax float64
 	replicaCount := s.metrics.ReplicaCount.Value()
 	bytesPerReplica := make([]float64, 0, replicaCount)
 	writesPerReplica := make([]float64, 0, replicaCount)
 	rankingsAccumulator := s.replRankings.newAccumulator()
+
+	// Query the current L0 sublevels and record the updated maximum to metrics.
+	s.l0SublevelsTracker.Lock()
+	s.l0SublevelsTracker.swag.Record(timeutil.Now(), float64(s.metrics.RdbL0Sublevels.Value()))
+	l0SublevelsMax, _ = s.l0SublevelsTracker.swag.Query(timeutil.Now())
+	s.l0SublevelsTracker.Unlock()
+
+	s.metrics.L0SubLevelsMax.Update(l0SublevelsMax)
```

Comment:
Isn't it odd that this happens as a byproduct of `Capacity`? I would've assumed this would remain in `updateEngineMetrics`.

Reply:
Updated to use this logic.